### DISCUSSION TITLE
Support for importing of Svelte packages?

### DISCUSSION THREAD

**Discussion by hskiba** [livebud/bud#99] [Q&A]
I'm new to Svelte and am trying to import a Svelte package for use in my project. 

For example, in one of my svelte files, I have:

```svelte
<script>
  export let stories = []
  import DataTable, { Head, Body, Row, Cell } from '@smui/data-table';
</script>
```

And when I load the page, I get the following.

```shell-script
$ bud run --port 3001
| Listening on http://127.0.0.1:3001
view: open error conjure: serve "bud/node_modules/@smui/data-table" > ✘ [ERROR] No loader is configured for ".svelte" files: node_modules/@smui/data-table/dist/DataTable.svelte

    node_modules/@smui/data-table/dist/index.js:1:22:
      1 │ import DataTable from './DataTable.svelte';
        ╵                       ~~~~~~~~~~~~~~~~~~~~
```

Is this type of import not yet supported?

**Comment by matthewmueller** [comment]
Ah interesting. I think the problem is that we don't support compiling `.svelte` files from within `node_modules`. The expectation is that `@smui/data-table/DataTable.svelte` has been transpiled to JS before publishing. 

This decision comes from the days where people were publishing all these packages that required a certain configurations of Babel to use. It was a big mess for the ecosystem.

I'd be curious why https://github.com/hperrin/svelte-material-ui/ does this. Do you mind opening an issue there? Might make sense to re-visit this decision depending on what they say.

**Reply by matthewmueller** [reply]
Also did you try this in any other framework like SvelteKit?

**Reply by hskiba** [reply]
> Also did you try this in any other framework like SvelteKit?

Yes, it seems to work using https://github.com/hperrin/smui-example-sveltekit/ as an example.

> I'd be curious why https://github.com/hperrin/svelte-material-ui/ does this. Do you mind opening an issue there? Might make sense to re-visit this decision depending on what they say.

I'm not sure I know what to ask here as I don't have an example of what we're expecting, but I'll try to figure this out...
